tests/functional/test_upgrade_cluster_imports.py

```diff
@@ -0,0 +1,180 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=W0611, W0621
+import coreapi
+import pytest
+
+from adcm_client.objects import ADCMClient
+from adcm_pytest_plugin.utils import get_data_dir
+from tests.library import errorcodes as err
+
+
+def test_upgrade_cluster_with_import(sdk_client_fs: ADCMClient):
+    """Scenario:
+    1. Create cluster for upgrade with exports
+    2. Create upgradable cluster with imports
+    3. Bind service and cluster
+    4. Upgrade cluster
+    5. Check that cluster was upgraded
+    """
+    bundle = sdk_client_fs.upload_from_fs(get_data_dir(
+        __file__, 'upgrade_cluster_with_export'))
+    bundle_import = sdk_client_fs.upload_from_fs(get_data_dir(
+        __file__, 'upgradable_cluster_with_import'))
+    cluster = bundle.cluster_create("test")
+    service = cluster.service_add(name="hadoop")
+    cluster_config_before = cluster.config()
+    service_config_before = service.config()
+    cluster_import = bundle_import.cluster_create("cluster_import")
+    cluster_import.bind(service)
+    cluster_import.bind(cluster)
+    upgr = cluster.upgrade(name='upgrade to 1.6')
+    upgr.do()
+    cluster.reread()
+    service.reread()
+    cluster_config_after = cluster.config()
+    service_config_after = service.config()
+    assert cluster.prototype().version == '1.6'
+    assert service.prototype().version == '2.2'
+    for variable in cluster_config_before:
+        assert cluster_config_before[variable] == cluster_config_after[variable]
+    for variable in service_config_before:
+        assert service_config_before[variable] == service_config_after[variable]
+
+
+def test_upgrade_cluster_with_export(sdk_client_fs: ADCMClient):
+    """Scenario:
+    1. Create cluster for upgrade with export
+    2. Create cluster for upgrade with import
+    3. Load upgradable bundle with import
+    4. Bind service and cluster
+    5. Upgrade cluster with import
+    6. Check that cluster was upgraded
+    """
+    bundle = sdk_client_fs.upload_from_fs(get_data_dir(__file__, 'cluster_with_export'))
+    bundle_import = sdk_client_fs.upload_from_fs(get_data_dir(
+        __file__, 'upgrade_cluster_with_import'))
+    sdk_client_fs.upload_from_fs(get_data_dir(__file__, 'upgradable_cluster_with_import'))
+    cluster = bundle.cluster_create("test")
+    service = cluster.service_add(name="hadoop")
+    cluster_import = bundle_import.cluster_create("cluster_import")
+    cluster_import.bind(service)
+    cluster_import.bind(cluster)
+    upgr = cluster_import.upgrade(name='upgrade to 1.6')
+    id_before = cluster_import.prototype_id
+    upgr.do()
+    cluster_import.reread()
+    assert cluster_import.prototype().version == '1.6'
+    assert cluster_import.prototype_id != id_before
+
+
+def test_incorrect_import_strcit_version(sdk_client_fs: ADCMClient):
```

Comment:
I think it is simple to parametrize that test with bundle dirs for all variants of tests with <, <=, >, >=. You just need a copy of the same bundle.